src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1101,6 +1102,8 @@ public void joinTokenRing(boolean finishJoiningRing,
             {
                 logger.warn("Some data streaming failed. Use nodetool to check bootstrap state and resume. For more, see `nodetool help bootstrap`. {}", SystemKeyspace.getBootstrapState());
             }
+
+            StorageProxy.instance.initialLoadPartitionDenylist();
```

Comment:
This happens relatively early in the grand scheme of things - see `CassandraDaemon.setup` and the call to `StorageService.instance.initServer` that cascades to here. So we'll load up our deny list prior to waiting for gossip to settle, prior to auth setup, and prior to re-enabling auto compaction. i.e. early enough in the init we shouldn't be exposing ourselves to a race with people accessing partitions we want to deny access to, but late enough that we're not relying on any systems or infra being up and running that might not be there.

Make sense? Also - full disclosure, I didn't do the initial implementation and it's been in production use for some time with this init ordering and running at scale, so that should factor in (both for and against) our degree of confidence the code.

Reply:
Ok, if that works for someone then we should be good. My point is that I cannot really validate this part because I do not understand the startup sequence very well.